### ISSUE TITLE
Grouping parameter widgets into accordions, tabs etc

### ISSUE THREAD

**Issue by jlstevens** [holoviz/panel#1632]
One of my favorite uses of panel is to render banks of param widgets. I would love to be able to set parameter precedence ranges to define accordions, tabs or whatever other type of grouping is available.

One API I can imagine might be something along the lines of:

`pn.Param(obj, accordion_range={'group1':(0,1), 'group2':(1,1.5)})`
or
`pn.Param(obj, tab_range={'group1':(0,1), 'group2':(1,1.5)})`

Negative precedences should not be allowed as those are supposed to be hidden. There may be a nicer alternative to this dictionary format but I do think having names for the groups will often be important.

**Comment by jbednar** [comment]
Sounds good. I'd also want a grouping that doesn't depend on accordions or tabs, but simply groups on the screen, either with a little box around them and a name on that group, or simply draws a separator between the groups.  It would be nice to be able to enable or not enable showing a name for each group of parameters.

> Negative precedences should not be allowed as those are supposed to be hidden. 

I wouldn't treat negative values specially. The threshold for displaying them is configurable, and one may wish to group even the non-displayed options, so that if the threshold is changed to display them (e.g. "expert" parameters), they will be grouped as desired.

**Comment by jlstevens** [comment]
Really this is about easily grouping parameters as widgets in various ways...

**Comment by philippjfr** [comment]
I'd suggest `groups` which accepts a list of tuples of the form `(Title, precedence_lower, precedence_higher)` or `(precedence_lower, precedence_higher)` and then a `group_layout` which is one of 'tabs', 'accordion', or 'column' (with separators).

**Comment by jbednar** [comment]
What happens to parameters not in an explicit group, i.e. with precedences that don't fall into any of the groups?  Presumably that would also be a group (e.g. a tab), but I don't know what it would be titled.

And if nothing falls into such a group, will there be no tabs, or one tab (for the default group)?

**Comment by jlstevens** [comment]
You could apply validation to make sure all parameters are covered by the supplied ranges, making those issues moot...

**Comment by jbednar** [comment]
I don't think it would make the issues moot, because the code defining the parameters and their preferences may be very distant from the code specifying the grouping, making the GUI code very vulnerable to breakage if some distant codebase defines a new parameter with an unusual precedence.  Adding such a parameter blindly is likely to occur in practice, because it won't cause any problems in the code where it's added, and nothing about that code will inherently document that the (distant) GUI is using precedences in that way.  So I think it's important for the behavior to be reasonable (if not necessarily precisely what's wanted) in cases where some parameters fall outside the grouping ranges.

**Comment by jlstevens** [comment]
When I say 'range' I am considering semi-bounded ranges e.g the last range could be `(4.5, None)` for instance to say 'parameters with precedences greater than 4.5 belong to this group'.
